exatomic/gaussian/output.py

```diff
@@ -608,6 +619,212 @@ def parse_momatrix(self):
+        #self.frequency['dx'] *= Length['Angstrom', 'au']
+        #self.frequency['dy'] *= Length['Angstrom', 'au']
+        #self.frequency['dz'] *= Length['Angstrom', 'au']
+
+    def parse_gradient(self):
+        # gradient regex
+        _regrad = 'Cartesian Gradient'
+        _reznums = 'Atomic numbers'
+        _renat = 'Number of atoms'
+
+        found = self.find(_regrad)
+        if not found:
+            return
+        else:
+            found = self.find(_regrad, _reznums, _renat, keys_only=True)
+        # get number of atoms
+        nat = self._intme(found[_renat])
+        # get atomic numbers
+        znums = self._dfme(found[_reznums], nat)
+        # generate symbols
+        symbols = list(map(lambda x: z2sym[x], znums))
+        # generate labels
+        label = [i for i in range(len(znums))]
+        ngrad = nat * 3
+        # get gradients(Ha/Bohr)
+        grad = self._dfme(found[_regrad], ngrad)
+        grad[abs(grad) < self.tol] = 0
+        # get x, y, z gradients
+        fx = grad[::3]
+        fy = grad[1::3]
+        fz = grad[2::3]
+        nframes = int(len(fx)/nat)
+        # generate frame labels
+        frame = [i for i in range(nframes)]
+        # extend to match sizes of all arrays
+        frame = np.repeat(frame, nat)
+        label = np.tile(label, nframes)
+        znums = np.tile(znums, nframes)
+        symbols = np.tile(symbols, nframes)
+        # create dataframe
+        self.gradient = pd.DataFrame.from_dict({"Z": znums, "label": label, "fx": fx, "fy": fy,
+                                                "fz": fz, "symbol": symbols, "frame": frame})
+
+    def parse_shielding_tensor(self):
+        # nmr shielding regex
+        _renmr = 'NMR shielding'
+        # base properties
+        _renat = 'Number of atoms'
+        _reznums = 'Atomic numbers'
+
+        found = self.find(_renmr)
+        if not found:
+            return
+        else:
+            found = self.find(_renmr, _renat, _reznums, keys_only=True)
+        # get number of atoms
+        nat = self._intme(found[_renat])
+        # get atomic numbers
+        znums = self._dfme(found[_reznums], nat)
+        # generate symbols
+        symbols = list(map(lambda x: z2sym[x], znums))
+        # generate labels
+        label = [i for i in range(len(znums))]
+        # get NMR shielding tensors (Hz)
+        # TODO: Check that this is in fact in Hz
+        shield = self._dfme(found[_renmr], nat * 9)
+        shield[abs(shield) < self.tol] = 0
+        # arrange into x, y, z cols
+        x = shield[::3]
+        y = shield[1::3]
+        z = shield[2::3]
+#        # conversion from Hz to ppm
+#        # (only done for some test calculations may not be accurate for all)
+#        x *= 1e6/18778.86
+#        y *= 1e6/18778.86
+#        z *= 1e6/18778.86
+        matrix = np.transpose([x,y,z]).reshape(nat,3,3)
+        matrix = np.asarray([0.5*(matrix[i] + np.transpose(matrix[i])) for i in range(len(matrix))])
```

Comment:
Since data is coming in as a list, I don't think `asarray` offers an advantage here over `array` (a copy will be required). You could try out something like `np.fromiter(0.5*(mat+np.transpose(mat)) for mat in matrix))`

Reply:
I'll try that. I know that I was doing this quite a bit and I must have forgotten to go back and change this one.

Reply:
I ended up trying this
```
matrix = np.fromiter(0.5 * (mat + np.transpose(mat)) for mat in matrix)
```
But it needed a dtype parameter so I added that and [ ] for the for loop and it gave back the error
```
ValueError: setting an array element with a sequence.
```